api/v1alpha1/api.go

```diff
@@ -43,22 +46,21 @@ type LLMRouteSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self.schema == 'OpenAI'"
 	APISchema LLMAPISchema `json:"inputSchema"`
-	// TargetRefs are the names of the Gateway resources this policy is being attached to.
+	// HTTPRouteRef is the name of the HTTPRoute resource that the Gateway will use to route the traffic.
 	// The namespace is "local", i.e. the same namespace as the LLMRoute.
 	//
-	// +optional
-	// +kubebuilder:validation:MaxItems=128
-	TargetRefs []gwapiv1a2.LocalPolicyTargetReferenceWithSectionName `json:"targetRefs"`
-	// BackendRefs lists the LLMBackends that this LLMRoute will route traffic to.
-	// The namespace is "local", i.e. the same namespace as the LLMRoute.
+	// In the matching configuration of the referenced HTTPRoute, `x-envoy-ai-gateway-llm-model` header
+	// can be used to describe the routing behavior.
 	//
-	// +kubebuilder:validation:MaxItems=128
-	BackendRefs []LLMBackendLocalRef `json:"backendRefs,omitempty"`
```

Comment:
If we follow this approach do we need LLMBackend Route? we can utilize the Backend CR coming from gateway itself. and missing backend apischema can handle by checking the LLMModelHeaderKey from Httproute.

Reply:
not only the schema but another backend-specific configs are needed anyways - rate limit, upstream authz, etc.